docker-compose.yml

```diff
@@ -2,26 +2,30 @@ notaryserver:
   build: .
   dockerfile: server.Dockerfile
   links:
-   - notarymysql
-   - notarysigner
-  ports:
-   - "8080"
-   - "4443:4443"
+    - notarymysql
+    - notarysigner
   environment:
-   - SERVICE_NAME=notary
-  command: -config=fixtures/server-config.json
+    - SERVICE_NAME="notaryserver"
+  ports:
+    - "8080"
+    - "4443:4443"
+  entrypoint: /bin/bash
+  command: -c "./migrations/migrate.sh && notary-server -config=fixtures/server-config.json"
 notarysigner:
-  volumes:
-   - /dev/bus/usb/003/010:/dev/bus/usb/002/010
-   - /var/run/pcscd/pcscd.comm:/var/run/pcscd/pcscd.comm
   build: .
   dockerfile: signer.Dockerfile
   links:
-   - notarymysql
-  command: -config=fixtures/signer-config.json
+    - notarymysql
+  environment:
+    - SERVICE_NAME="notarysigner"
+  entrypoint: /bin/bash
+  command: -c "./migrations/migrate.sh && notary-signer -config=fixtures/signer-config.json"
 notarymysql:
   volumes:
-    - notarymysql:/var/lib/mysql
-  build: ./notarymysql/
+    - ./notarymysql/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
+    - ./notarymysql/data:/var/lib/mysql
```

Comment:
Changing this to "notarymysql:/var/lib/mysql" fixed the bug for me where I was getting all sorts of permissions errors.

If I ran it without this change, then made this change, then ran build/up again, for some reason everything was still broken.  So did `docker-compose rm` and removed the `notary_notarymysql` image, also did `docker volume rm notarymysql`.

Then I built with master, ran some repo inits and publishes to add data, then switched back to this branch, made the change, and `docker-compose build && docker-compose up` again, and minimal errors (at least, not container-stopping errors).

It looks like it's still also not running migrations for me though, but since the table looks like it should, maybe they just aren't run?  Will try again with an empty DB.

```
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Using mutexes to ref count buffer pool pages
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: The InnoDB memory heap is disabled
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Memory barrier is not used
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Compressed tables use zlib 1.2.8
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Using Linux native AIO
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Using SSE crc32 instructions
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Initializing buffer pool, size = 256.0M
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Completed initialization of buffer pool
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Highest supported file format is Barracuda.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: The log sequence numbers 1642992 and 1642992 in ibdata files do not match the log sequence number 1655569 in the ib_logfiles!
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Database was not shutdown normally!
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Starting crash recovery.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Reading tablespace information from the .ibd files...
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Restoring possible half-written data pages 
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: from the doublewrite buffer...
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Creating tablespace and datafile system tables.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Tablespace and datafile system tables created.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Warning] InnoDB: Resizing redo log from 2*320 to 2*3072 pages, LSN=1659978
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Warning] InnoDB: Starting to delete and rewrite log files.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Setting log file ./ib_logfile101 size to 48 MB
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Setting log file ./ib_logfile1 size to 48 MB
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Warning] InnoDB: New log files created, LSN=1660428
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: 128 rollback segment(s) are active.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB: Waiting for purge to start
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.26-76.0 started; log sequence number 1655569
notarymysql_1  | 2016-02-05 22:23:34 140579962849024 [Note] InnoDB: Dumping buffer pool(s) not yet started
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] Plugin 'FEEDBACK' is disabled.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] Server socket created on IP: '::'.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Warning] 'user' entry 'root@d23a73ed736f' ignored in --skip-name-resolve mode.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Warning] 'proxies_priv' entry '@% root@d23a73ed736f' ignored in --skip-name-resolve mode.
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [ERROR] Missing system table mysql.roles_mapping; please run mysql_upgrade to create it
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [ERROR] Incorrect definition of table mysql.event: expected column 'sql_mode' at position 14 to have type set('REAL_AS_FLOAT','PIPES_AS_CONCAT','ANSI_QUOTES','IGNORE_SPACE','IGNORE_BAD_TABLE_OPTIONS','ONLY_FULL_GROUP_BY','NO_UNSIGNED_SUBTRACTION','NO_DIR_IN_CREATE','POSTGRESQL','ORACLE','MSSQL','DB2','MAXDB','NO_KEY_OPTIONS','NO_TABLE_OPTIONS','NO_FIELD_OPTIONS','MYSQL323','MYSQL40','ANSI','NO_AUTO_VALUE_ON_ZERO','NO_BACKSLASH_ESCAPES','STRICT_TRANS_TABLES','STRICT_ALL_TABLES','NO_ZERO_IN_DATE','NO_ZERO_DATE','INVALID_DATES','ERROR_FOR_DIVISION_BY_ZERO','TRADITIONAL','NO_AUTO_CREATE_USER','HIGH_NOT_PRECEDENCE','NO_ENGINE_SUBSTITUTION','PAD_CHAR_TO_FULL_LENGTH'), found type set('REAL_AS_FLOAT','PIPES_AS_CONCAT','ANSI_QUOTES','IGNORE_SPACE','NOT_USED','ONLY_FULL_GROUP_BY','NO_UNSIGNED_SUBTRACTION','NO_DIR_IN_CREATE','POSTGRESQL','ORACLE','MSSQL','DB2','MAXDB','NO_KEY_OPTIONS','NO_TABLE_OPTIONS','NO_FIELD_OPTIONS','MYSQL323','MYSQL40','ANSI','NO_AUTO_VALUE_ON_ZERO','NO_BACKSLASH_ESCAPES','STRICT_TRANS_TABLES','STRICT_A
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [ERROR] mysqld: Event Scheduler: An error occurred when initializing system tables. Disabling the Event Scheduler.
notarymysql_1  | 2016-02-05 22:23:34 140580748887808 [Warning] Failed to load slave replication state from table mysql.gtid_slave_pos: 1146: Table 'mysql.gtid_slave_pos' doesn't exist
notarymysql_1  | 2016-02-05 22:23:34 140580749445056 [Note] mysqld: ready for connections.
notarymysql_1  | Version: '10.1.10-MariaDB-1~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
notarymysql_1  | 2016-02-05 22:24:46 7fdb81906700 InnoDB: Error: Table "mysql"."innodb_table_stats" not found.
```